db/migrate/20230515210900_create_annotation_references.rb

```diff
@@ -0,0 +1,36 @@
+class CreateAnnotationReferences < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction! if System::Database.postgres?
+
+  def change
+    if System::Database.mysql?
+      options = "CHARSET=utf8mb4 COLLATE=utf8mb4_bin"
+    end
+
+    create_table :annotations, options: options do |t|
+      t.string :name, limit: 42, null: false
+      t.string :value
```

Comment:
Storing structured data in annotations. You sure. Do you have an example software that uses this? Or in case you can imagine a good use case. I'm concerned about abusing the feature.